### PR TITLE
 Add envoy gateway testing

### PR DESCRIPTION
**Pull request by dprotaso**
Part of https://github.com/knative-extensions/net-gateway-api/issues/566

### CODE REVIEW

File: hack/test-env.sh [hunk 1]
```diff
@@ -19,3 +19,6 @@ export ISTIO_VERSION="1.22.0"
 export ISTIO_UNSUPPORTED_E2E_TESTS="retry,httpoption"
 export CONTOUR_VERSION="v1.29.0"
 export CONTOUR_UNSUPPORTED_E2E_TESTS="httpoption"
+
+export ENVOY_GATEWAY_VERSION="latest"
+export ENVOY_GATEWAY_UNSUPPORTED_E2E_TESTS="httpoption"
```
Comment (arkodg) [inline hack/test-env.sh:24]: lets disable host rewrite for now ? with a comment saying EG doesnt support `ExternalName` Service routing ?

**Comment by dprotaso** [comment]
gateway isn't getting and addresss

```
    conditions:
    - lastTransitionTime: "2024-06-03T21:22:55Z"
      message: The Gateway has been scheduled by Envoy Gateway
      observedGeneration: 1
      reason: Accepted
      status: "True"
      type: Accepted
    - lastTransitionTime: "2024-06-03T21:22:55Z"
      message: No addresses have been assigned to the Gateway
      observedGeneration: 1
      reason: AddressNotAssigned
      status: "False"
      type: Programmed
```

**Comment by arkodg** [comment]
hey @dprotaso can we kick CI once, to see if this is a flake or something that just doesnt work in this setup ? setting the Gateway.Status.Addresses from the Svc is something that is verified in the conformance tests, so not sure why its not working here

**Comment by dprotaso** [comment]
I also repro’d locally I don’t think it’s a fake - see the setup in the
third_party folder

**Comment by dprotaso** [comment]
/test integration-tests-envoy-gateway_net-gateway-api_main
/test integration-tests-envoy-gateway-conformance_net-gateway-api_main

I believe the failure is GKE related - similar to: https://github.com/knative-extensions/net-gateway-api/issues/708

**Comment by dprotaso** [comment]
@arkodg all good now

/assign @izabelacg 

cc @ReToCode @skonto

**Pull request merged**